sys/linux/socket_tipc_netlink.txt

```diff
@@ -93,6 +93,12 @@ tipc_name_table_query {
 	upbound		int32be
 }
 
+tipc_aead_key {
+	alg_name	array[int8, 32]
```

Comment:
Thanks! I've changed the code correspondingly. I am wondering if the current way is proper though, as I'm afraid that it will force the field to be always set to "gcm(aes)"

Reply:
There are already types defined for the crypto algorithm strings in `sys/linux/socket_alg.txt`.

`alg_aead_name` **would** be the right one to use.  However it has a different length: 64 bytes instead of the 32 that is expected here.  I'm not sure how to best handle this.  Perhaps define a type `alg_aead_name_32` in `sys/linux/socket_alg.txt` alongside `alg_aead_name`?

Reply:
Hmm so I just tried that and that would require "include tipc.h" in the socket_alg.txt file so that we have access to TIPC_AEAD_ALG_NAME. I'm wondering if that's what we want. Alternatively we can just hard code it as 32 but it just doesn't sound right to me.

Reply:
I guess we don't care too much about making syzkaller specify any arbitrary crypto API algorithm for this, since the kernel checks that it is "gcm(aes)" and returns an error otherwise.  So what you did is okay for now.  We'll need to remember to update this if/when more algorithms are added later though.